engineyard/ey-cookbooks-stable-v5
# [YT-CC-1191] Nullmailer package installation and configuration cookbook added

## Description of your patch

Nullmailer installation and configuration cookbook added to stable-v5 Chef.
SSMTP replacement with Nullmailer capability added.

## Recommended Release Notes

Capability to use Nulmailer instead of SSMTP mail service

## Estimated risk

Low
New functionality will not affect running stacks. It only adds capability to switch to Nullmailer if they want to. 

## Components involved

ey-init cookbook
Nullmailer package
SSMTP package

## Description of testing done

See QA instructions

## QA Instructions

1. Create simple Ruby application.  (https://github.com/engineyard/todo.git) can be used for this testing.
2. Boot the environment with QA stack created from the CC-1191 branch. Environment should be run with following parameters:
  * Environment type - Single Instance.
3. Run command `eix ssmtp`, to ensure that SSMTP service installed.
4. Update `/etc/chef/dna.json file`. Find JSON Hash value:
``` 
            "components": [
              {
                "key": "ssmtp"
              },
```
and replace it with
```
            "components": [
              {
                "key": "nullmailer"
              },
```
5. Run Chef `PATH=/usr/local/ey_resin/bin:$PATH /home/ey/bin/chef-solo -j /etc/chef/dna.json -c /etc/chef/solo.rb`
6. Run command `eix ssmtp`. Ensure package mail-mta/ssmtp not installed.
7. Run command `eix nullmailer`. Ensure package mail-mta/nullmailer is installed.
8. Run command `/etc/init.d/nullmailer status`. Ensure service is in "started" state.
